python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -629,6 +645,85 @@ def unpack(
         check.failed(f"Unexpected class {self.klass}")
 
 
+@whitelist_for_serdes
+class SkipOnNotAllParentsUpdatedRule(
+    AutoMaterializeRule,
+    NamedTuple(
+        "_SkipOnNotAllParentsUpdatedRule", [("require_update_on_all_parent_partitions", bool)]
+    ),
+):
+    """An auto-materialize rule that enforces that an asset can only be requested for execution
+    if all parents have been materialized since its last materialization.
+
+    Properties:
+        require_update_on_all_parent_partitions (Optional[bool]): If true, requires all upstream
+            partitions to be materialized in order for a given asset to update. If false, requires
```

Comment:
materialized since the asset's last materialization?

Reply:
also, this is only relevant in the case that an asset or asset partition depends on multiple upstream asset partitions, right? we should clarify that.

Reply:
if a downstream asset partition depends on some partitions, we'll never wait on partitions that it doesn't depend on to update it

Reply:
sounds good, updated